### PR TITLE
feat(@spartacus/storefront): use generic page layout for 404

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
closes #1252

### CODE REVIEW

File: projects/storefrontstyles/scss/layout/_index.scss [hunk 1]
```diff
@@ -33,3 +33,7 @@ cx-page-layout {
 .LoginPageTemplate {
   @extend %LoginPageTemplate !optional;
 }
+
+.ErrorPageTemplate {
+  @extend %LoginPageTemplate !optional;
```
Comment (Platonn) [inline projects/storefrontstyles/scss/layout/_index.scss]: Why .ErrorPageTemplate is extending %LoginPageTemplate?

File: projects/storefrontlib/src/lib/ui/pages/404/404.component.html [hunk 2]
```diff
@@ -1,20 +0,0 @@
-<div class="cx-page">
```
Comment (Platonn) [inline projects/storefrontlib/src/lib/ui/pages/404/404.component.html]: Why do we remove this component implementation? Shouldn't it be just transformed to a CMS component?

File: projects/storefrontstyles/scss/layout/_index.scss (hunk 1 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontstyles/scss/layout/_index.scss]: similar to the other https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/1249#discussion_r255499150:

> Yes, its a bit in between situation. Once all page templates are moved to this approach, i'm planning on eliminating backend templates. The front-end will probably lead, and AccountPageTemplate, LoginPageTemplate, NotFound, etc. are going to be merged. I need to come with a good name, and then it's time to move. I'm delaying this decision, first want to have good visibility on all of them.

Please bare with me. I'll do an update once we've passed all layouts.

File: projects/storefrontlib/src/lib/ui/pages/404/404.component.html (hunk 2 above)
Comment (tobi-or-not-tobi) [reply projects/storefrontlib/src/lib/ui/pages/404/404.component.html]: There is already a component in the backend. But fair point, I've created #1254 in order to have the same UI back (but then CMS driven).

**Pull request merged**